tests/test_vector3.h

```diff
@@ -0,0 +1,717 @@
+			"to_diagonal_matrix() must return expected value.");
+
+	CHECK_MESSAGE(
+			Basis(3, 0, 0, 0, -2, 0, 0, 0, 1).is_equal_approx(test_subject_c.to_diagonal_matrix()),
+			"to_diagonal_matrix() must return expected value.");
+}
+
+TEST_CASE("[Vector3] Abs") {
+	const Vector3 test_subject_a = Vector3(1.0, 2.0, 3.0);
+	const Vector3 test_subject_b = Vector3(-1.0, 3.0, -2.0);
+	const Vector3 test_subject_c = Vector3(3.0, -2.0, 1.0);
+
+	CHECK_MESSAGE(
+			Vector3(1, 2, 3).is_equal_approx(test_subject_a.abs()),
+			"abs() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(1, 3, 2).is_equal_approx(test_subject_b.abs()),
+			"abs() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(3, 2, 1).is_equal_approx(test_subject_c.abs()),
+			"abs() must return expected value.");
+}
+
+TEST_CASE("[Vector3] Floor") {
+	const Vector3 test_subject_a = Vector3(1.3, 2.7, 3.4);
+	const Vector3 test_subject_b = Vector3(-1.2, 3.9, -2.9);
+	const Vector3 test_subject_c = Vector3(3.7, -2.3, 1.5);
+
+	CHECK_MESSAGE(
+			Vector3(1, 2, 3).is_equal_approx(test_subject_a.floor()),
+			"floor() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(-2, 3, -3).is_equal_approx(test_subject_b.floor()),
+			"floor() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(3, -3, 1).is_equal_approx(test_subject_c.floor()),
+			"floor() must return expected value.");
+}
+
+TEST_CASE("[Vector3] Sign") {
+	const Vector3 test_subject_a = Vector3(1.3, 2.7, 3.4);
+	const Vector3 test_subject_b = Vector3(-1.2, 3.9, -2.9);
+	const Vector3 test_subject_c = Vector3(3.7, -2.3, 1.5);
+
+	CHECK_MESSAGE(
+			Vector3(1, 1, 1).is_equal_approx(test_subject_a.sign()),
+			"sign() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(-1, 1, -1).is_equal_approx(test_subject_b.sign()),
+			"sign() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(1, -1, 1).is_equal_approx(test_subject_c.sign()),
+			"sign() must return expected value.");
+}
+
+TEST_CASE("[Vector3] Ceil") {
+	const Vector3 test_subject_a = Vector3(1.3, 2.7, 3.4);
+	const Vector3 test_subject_b = Vector3(-1.2, 3.9, -2.9);
+	const Vector3 test_subject_c = Vector3(3.7, -2.3, 1.5);
+
+	CHECK_MESSAGE(
+			Vector3(2, 3, 4).is_equal_approx(test_subject_a.ceil()),
+			"ceil() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(-1, 4, -2).is_equal_approx(test_subject_b.ceil()),
+			"ceil() must return expected value.");
+
+	CHECK_MESSAGE(
+			Vector3(4, -2, 2).is_equal_approx(test_subject_c.ceil()),
+			"ceil() must return expected value.");
+}
+
+TEST_CASE("[Vector3] Round") {
+	const Vector3 test_subject_a = Vector3(1.3, 2.7, 3.4);
+	const Vector3 test_subject_b = Vector3(-1.2, 3.9, -2.9);
+	const Vector3 test_subject_c = Vector3(3.7, -2.3, 1.5);
+
+	CHECK_MESSAGE(
+			Vector3(1, 3, 3).is_equal_approx(test_subject_a.round()),
+			"round() must return expected value.");
+
+	CAPTURE(test_subject_b.round());
```

Comment:
These `CAPTURE` statements are redundant in the context where they are used in this PR, should likely be removed.

Reply:
Fixed